packages/starknet-snap/src/rpcs/executeTxn.ts

```diff
@@ -0,0 +1,397 @@
+
+/**
+ *
+ * @param deployResp
+ * @param chainId
+ */
+async function recordAccountDeployment(
+  deployResp: DeployContractResponse, // Adjust type as needed
+  chainId: string,
+) {
+  const accountStateManager = new AccountStateManager(false);
+  const account = await accountStateManager.getAccount({
+    address: deployResp.contract_address,
+    chainId,
+  });
+  if (account === null) {
+    throw new Error('Account contract not found');
+  }
+  await accountStateManager.updateAccount({
+    ...account,
+    deployTxnHash: deployResp.transaction_hash,
+    upgradeRequired: false,
+    deployRequired: false,
+  });
+
+  const txn: Transaction = {
+    txnHash: deployResp.transaction_hash,
+    txnType: VoyagerTransactionType.DEPLOY_ACCOUNT,
+    chainId,
+    senderAddress: deployResp.contract_address,
+    contractAddress: deployResp.contract_address,
+    contractFuncName: '',
+    contractCallData: [],
+    finalityStatus: TransactionStatus.RECEIVED,
+    executionStatus: TransactionStatus.RECEIVED,
+    status: '',
+    failureReason: '',
+    eventIds: [],
+    timestamp: Math.floor(Date.now() / 1000),
+  };
+  const transactionStateManager = new TransactionStateManager();
+  await transactionStateManager.addTransaction(txn);
+}
+
+/**
+ * The RPC handler to execute a transaction.
+ */
+export class ExecuteTxnRpc extends AccountRpcController<
+  ExecuteTxnParams,
+  ExecuteTxnResponse
+> {
+  protected requestStruct = ExecuteTxnRequestStruct;
+
+  protected responseStruct = ExecuteTxnResponseStruct;
+
+  /**
+   * Execute the transaction request handler.
+   *
+   * @param params - The parameters of the request.
+   * @param params.address - The address of the signer.
+   * @param params.calls - The invoke calls to execute
+   * @param params.abis - The abis associated to invocations.
+   * @param params.details - The detail associated to the call.
+   * @returns The InvokeFunctionResponse as an `ExecuteTxnResponse`.
+   */
+  async execute(params: ExecuteTxnParams): Promise<ExecuteTxnResponse> {
+    return super.execute(params);
+  }
+
+  protected async handleRequest(
+    params: ExecuteTxnParams,
+  ): Promise<ExecuteTxnResponse> {
+    const { address, calls, abis, details } = params;
+    const isArray = Array.isArray(calls);
+    const callsArray = isArray ? calls : [calls];
+    const invocations: Invocations = callsArray.map((call) => {
+      return {
+        type: TransactionType.INVOKE,
+        payload: call,
+      };
+    });
+
+    const estimateFeeResp = await getEstimatedFees(
+      this.network,
+      address,
+      this.account.privateKey,
+      this.account.publicKey,
+      invocations as unknown as Invocations,
+      {
+        version: details?.version ?? TRANSACTION_VERSION,
+      },
+    );
+
+    const accountDeployed = !estimateFeeResp.includeDeploy;
+
+    if (
+      !(await this.getExecuteTxnConsensus(
+        address,
+        callsArray,
```

Comment:
```suggestion
        calls,
```

convert it to array or not is mainly for display, hence, lets leave the work in getExecuteTxnConsensus